ledger/trie/trie_test.go

```diff
@@ -0,0 +1,278 @@
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package trie_test
+
+import (
+	"encoding/hex"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/utils"
+	reference "github.com/onflow/flow-go/ledger/complete/mtrie/trie"
+
+	"github.com/optakt/flow-dps/ledger/store"
+	"github.com/optakt/flow-dps/ledger/trie"
+	"github.com/optakt/flow-dps/testing/helpers"
+	"github.com/optakt/flow-dps/testing/mocks"
+)
+
+func TestTrie_InsertLeftRegister(t *testing.T) {
+	store := mocks.BaselineStore()
+	trie := trie.NewEmptyTrie(store)
+
+	path := utils.PathByUint16LeftPadded(0)
+	payload := utils.LightPayload(11, 12345)
+
+	trie.Insert(path, payload)
+
+	expectedRootHashHex := "b30c99cc3e027a6ff463876c638041b1c55316ed935f1b3699e52a2c3e3eaaab"
+
+	got := trie.RootHash()
+	require.Equal(t, expectedRootHashHex, hex.EncodeToString(got[:]))
+}
+
+func TestTrie_InsertRightRegister(t *testing.T) {
+	store := mocks.BaselineStore()
+	trie := trie.NewEmptyTrie(store)
+
+	var path ledger.Path
+	for i := 0; i < len(path); i++ {
+		path[i] = uint8(255)
+	}
+	payload := utils.LightPayload(12346, 54321)
+
+	trie.Insert(path, payload)
+
+	expectedRootHashHex := "4313d22bcabbf21b1cfb833d38f1921f06a91e7198a6672bc68fa24eaaa1a961"
+
+	got := trie.RootHash()
+	require.Equal(t, expectedRootHashHex, hex.EncodeToString(got[:]))
+}
+
+func TestTrie_InsertMiddleRegister(t *testing.T) {
+	store := mocks.BaselineStore()
+	trie := trie.NewEmptyTrie(store)
+
+	path := utils.PathByUint16LeftPadded(56809)
+	payload := utils.LightPayload(12346, 59656)
+
+	trie.Insert(path, payload)
+
+	expectedRootHashHex := "4a29dad0b7ae091a1f035955e0c9aab0692b412f60ae83290b6290d4bf3eb296"
+
+	got := trie.RootHash()
+	require.Equal(t, expectedRootHashHex, hex.EncodeToString(got[:]))
+}
+
+func TestTrie_InsertManyRegisters(t *testing.T) {
+	dir, err := os.MkdirTemp("", "")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+
+	store, err := store.NewStore(zerolog.New(io.Discard), 4*1000*1000, dir)
+	require.NoError(t, err)
+
+	trie := trie.NewEmptyTrie(store)
+
+	paths, payloads := helpers.SampleRandomRegisterWrites(helpers.NewGenerator(), 12001)
+
+	for i := range paths {
+		trie.Insert(paths[i], &payloads[i])
+	}
+
+	got := trie.RootHash()
+	expectedRootHashHex := "74f748dbe563bb5819d6c09a34362a048531fd9647b4b2ea0b6ff43f200198aa"
+	assert.Equal(t, expectedRootHashHex, hex.EncodeToString(got[:]))
+}
+
+// FIXME: Make a mock store which can substitute the store and preserve functionality for tests.
+
+func TestTrie_InsertFullTrie(t *testing.T) {
+	dir, err := os.MkdirTemp("", "")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
```

Comment:
This block of code is used more than once -> it could be separated in a function for creating a directory in some helper file for the tests; as well as some default constants used for testing the package

Reply:
It's planned to create a helper for it yes :) Thanks for the reminder